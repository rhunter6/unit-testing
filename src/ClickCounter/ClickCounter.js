import React, { Component } from "react";

class ClickCounter extends Component {
  constructor(props) {
    super(props);
    this.state = { counter: 0 };
  }

  onClick() {
    this.setState({ counter: this.state.counter + 1 });
  }

  render() {
    return (
      <div>
        Click Count: {this.state.counter}
        <br />
        <br />
        <button onClick={() => this.onClick()}>Click Me!</button>
      </div>
    );
  }
}

export default ClickCounter;
