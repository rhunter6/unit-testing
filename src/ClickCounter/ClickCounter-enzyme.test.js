import { shallow } from "enzyme";
import ClickCounter from "./ClickCounter";


describe("ClickCounter testing", () => {
  it("should render", () => {
    const shallowWrapper = shallow(<ClickCounter />);
    expect(shallowWrapper).toMatchSnapshot();
  });

  it("should click the counter", () => {
    const shallowWrapper = shallow(<ClickCounter />);
    expect(shallowWrapper.state().counter).toBe(0);
    shallowWrapper.find("button").simulate('click');
    expect(shallowWrapper.state().counter).toBe(1);
    shallowWrapper.find("button").simulate('click');
    shallowWrapper.find("button").simulate('click');
    shallowWrapper.find("button").simulate('click');
    expect(shallowWrapper.state().counter).toBe(4);

    
    shallowWrapper.setState({counter: 25});
    expect(shallowWrapper.state().counter).toBe(25);
    expect(shallowWrapper).toMatchSnapshot();
  })
});
