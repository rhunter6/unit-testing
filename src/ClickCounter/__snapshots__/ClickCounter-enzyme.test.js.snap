// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClickCounter testing should click the counter 1`] = `
<div>
  Click Count: 
  25
  <br />
  <br />
  <button
    onClick={[Function]}
  >
    Click Me!
  </button>
</div>
`;

exports[`ClickCounter testing should render 1`] = `
<div>
  Click Count: 
  0
  <br />
  <br />
  <button
    onClick={[Function]}
  >
    Click Me!
  </button>
</div>
`;
